src/MainWindow.vala

```diff
@@ -214,15 +214,15 @@ namespace Terminal {
             /* This requires all restored tabs to be initialized first so that the shell location is available */
             /* Do not add a new tab if location is already open in existing tab */
             string? location = null;
-
             if (directory == null || directory == "") {
                 if (notebook.tabs.first () == null || command != null || create_new_tab) { //Ensure at least one tab
                     new_tab ("", command);
                 }
 
                 return;
             } else {
-                location = directory;
+                // Substitute "~" etc but do not add the "file://" scheme
+                location = Utils.sanitize_path (directory, Environment.get_current_dir (), false);
```

Comment:
this is wrong, `Environment.get_current_dir()` is the directory of the first invocation (in most of cases, $HOME), not of the current invocation.

for example, assuming terminal was already opened via dock or wingpanel, if the user is in `/home/user/projects/terminal/build` and run `io.elementary.terminal -t --working-directory=../src`, this will try to open `/home/src` not `/home/user/projects/terminal/src`.

ps: `-tw ../src` and `--working-directory ../src` will work as expected because they are handled by the shell, so i don't think there's much value on this change.

Reply:
OK i'll revert this for now - it can be addressed later if needed.